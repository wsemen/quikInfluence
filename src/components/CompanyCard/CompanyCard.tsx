import {
  Box,
  Divider,
  Flex,
  Heading,
  Text,
  useColorMode,
} from '@chakra-ui/react';
import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { faMoneyBillWaveAlt, faUser } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import quikColorConstants, {
  cardThemeColor,
  borderThemeColor,
} from 'utils/constants/colorConstants';

type CompanyCardProps = {
  companyName: string;
  companyLogo?: string;
  users: number;
  leads: number;
  revenue: number;
};

const CompanyCard = ({
  companyName,
  companyLogo,
  users,
  leads,
  revenue,
}: CompanyCardProps) => {
  const { colorMode } = useColorMode();
  return (
    <Flex
      padding={[6, 6]}
      background={cardThemeColor[colorMode]}
      width="100%"
      borderRadius="8px"
      flexWrap="wrap"
    >
      <Flex width="100%" marginBottom="15px">
        <Box
          height="48px"
          width="48px"
          bgColor={quikColorConstants.influenceRedWithOpacity}
          border={`1px solid ${borderThemeColor[colorMode]}`}
          marginRight="6px"
          borderRadius="10px"
        ></Box>
        <Text margin="auto 0" fontSize="2xl" letterSpacing="0.2px">
          {companyName}
        </Text>
      </Flex>
      <Divider />

      <Flex width="100%" marginTop="15px" justify="space-between">
        <Flex paddingX="1px">
          <FontAwesomeIcon
            style={{
              margin: 'auto',
            }}
            size="sm"
            icon={faUser as IconProp}
          />
          <Text paddingLeft="5px" fontSize="lg" margin="auto">
            {users} Users
          </Text>
        </Flex>

        <Flex paddingX="1px">
          <FontAwesomeIcon
            style={{
              margin: 'auto',
            }}
            size="sm"
            icon={faUser as IconProp}
          />
          <Text paddingLeft="5px" fontSize="lg" margin="auto">
            {leads} Leads
          </Text>
        </Flex>

        <Flex paddingX="1px">
          <FontAwesomeIcon
            style={{
              margin: 'auto',
            }}
            size="sm"
            icon={faMoneyBillWaveAlt as IconProp}
          />
          <Text paddingLeft="5px" fontSize="lg" margin="auto">
            {revenue}K
          </Text>
        </Flex>
      </Flex>
    </Flex>
  );
};

export default CompanyCard;
