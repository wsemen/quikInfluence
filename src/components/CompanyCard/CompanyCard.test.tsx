import { render } from "@testing-library/react";
import CompanyCard from ".";

it("renders CompanyCard component", () => {
  const { container } = render(
    <CompanyCard companyName={"Mocking Bird"} users={0} leads={0} revenue={0} />
  );
  expect(container).toMatchSnapshot();
});
