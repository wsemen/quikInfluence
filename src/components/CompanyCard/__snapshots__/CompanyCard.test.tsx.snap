// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CompanyCard component 1`] = `
<div>
  <div
    class="css-d6k11m"
  >
    <div
      class="css-vrmnwf"
    >
      <div
        class="css-jqpl9f"
      />
      <p
        class="chakra-text css-15gds93"
      >
        Mocking Bird
      </p>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="css-k3fo5g"
    >
      <div
        class="css-e2q3qv"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user fa-w-14 fa-sm "
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="margin: auto;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
          />
        </svg>
        <p
          class="chakra-text css-5yrxju"
        >
          0
           Users
        </p>
      </div>
      <div
        class="css-e2q3qv"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user fa-w-14 fa-sm "
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="margin: auto;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
          />
        </svg>
        <p
          class="chakra-text css-5yrxju"
        >
          0
           Leads
        </p>
      </div>
      <div
        class="css-e2q3qv"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-money-bill-wave-alt fa-w-20 fa-sm "
          data-icon="money-bill-wave-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="margin: auto;"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M621.16 54.46C582.37 38.19 543.55 32 504.75 32c-123.17-.01-246.33 62.34-369.5 62.34-30.89 0-61.76-3.92-92.65-13.72-3.47-1.1-6.95-1.62-10.35-1.62C15.04 79 0 92.32 0 110.81v317.26c0 12.63 7.23 24.6 18.84 29.46C57.63 473.81 96.45 480 135.25 480c123.17 0 246.34-62.35 369.51-62.35 30.89 0 61.76 3.92 92.65 13.72 3.47 1.1 6.95 1.62 10.35 1.62 17.21 0 32.25-13.32 32.25-31.81V83.93c-.01-12.64-7.24-24.6-18.85-29.47zM320 352c-44.19 0-80-42.99-80-96 0-53.02 35.82-96 80-96s80 42.98 80 96c0 53.03-35.83 96-80 96z"
            fill="currentColor"
          />
        </svg>
        <p
          class="chakra-text css-5yrxju"
        >
          0
          K
        </p>
      </div>
    </div>
  </div>
</div>
`;
